src/git/config.go

```diff
@@ -61,7 +61,11 @@ func GetParentBranchMap() map[string]string {
 	for _, key := range getConfigurationKeysMatching("^git-town-branch\\..*\\.parent$") {
 		child := strings.TrimSuffix(strings.TrimPrefix(key, "git-town-branch."), ".parent")
 		parent := GetConfigurationValue(key)
-		result[child] = parent
+		if HasBranch(child) && HasBranch(parent) {
+			result[child] = parent
+		} else {
+			removeConfigurationValue(key)
```

Comment:
This is beautifully simple. My concern here is that a function called `GetParentBranchMap` sounds pure and fast, but it now has a pretty hefty side effect of deleting information on disk when called. This impacts maintainability and the principle of least surprise.

I would be willing to pay a small complexity tax in order to separate these two concerns better. 

* add a second return value to `GetParentBranchMap` that returns the outdated entries. The first return value of `GetParentBranchMap` would be the same `map[string]string` as it is now, with only current branches. The second one would be a `[]string` containing outdated entries.

* create a `RemoveOutdatedBranches` function that takes the second `[]string` return value from `GetParentBranchMap` and removes the entries from disk. This method would be called after calling `GetParentBranchMap`.

```go
func GetParentBranchMap() (branchMap map[string]string, outdated []string) { 
  ...
  if HasBranch(child) && HasBranch(parent) {
	result[child] = parent
  } else {
    outdated = append(outdated, key)
}

func RemoveOutdatedBranches([]string outdated) {
  for _, key := range outdated {
    removeConfigurationValue(key)
  }
}

Reply:
`GetParentBranchMap` may actually call two git commands to read the config file and to get the list of current branches so its not that pure and fast. In the worst case, where there a lot of entries to be removed, then yes, it can cause a slowdown. 

I agree extracting makes it less surprising.